editor/plugins/polygon_2d_editor_plugin.h

```diff
@@ -50,50 +50,59 @@ class VScrollBar;
 class Polygon2DEditor : public AbstractPolygon2DEditor {
 	GDCLASS(Polygon2DEditor, AbstractPolygon2DEditor);
 
-	enum Mode {
-		MODE_EDIT_UV = MODE_CONT,
-		UVEDIT_POLYGON_TO_UV,
-		UVEDIT_UV_TO_POLYGON,
-		UVEDIT_UV_CLEAR,
-		UVEDIT_GRID_SETTINGS
+	enum {
+		MENU_POLYGON_TO_UV,
+		MENU_UV_TO_POLYGON,
+		MENU_UV_CLEAR,
+		MENU_GRID_SETTINGS,
 	};
 
-	enum UVMode {
-		UV_MODE_CREATE,
-		UV_MODE_CREATE_INTERNAL,
-		UV_MODE_REMOVE_INTERNAL,
-		UV_MODE_EDIT_POINT,
-		UV_MODE_MOVE,
-		UV_MODE_ROTATE,
-		UV_MODE_SCALE,
-		UV_MODE_ADD_POLYGON,
-		UV_MODE_REMOVE_POLYGON,
-		UV_MODE_PAINT_WEIGHT,
-		UV_MODE_CLEAR_WEIGHT,
-		UV_MODE_MAX
+	enum Mode {
+		MODE_POINTS,
+		MODE_POLYGONS,
+		MODE_UV,
+		MODE_BONES,
+		MODE_MAX
 	};
 
-	Button *uv_edit_mode[4];
-	Ref<ButtonGroup> uv_edit_group;
+	enum Action {
+		ACTION_CREATE,
+		ACTION_CREATE_INTERNAL,
+		ACTION_REMOVE_INTERNAL,
+		ACTION_EDIT_POINT,
+		ACTION_MOVE,
+		ACTION_ROTATE,
+		ACTION_SCALE,
+		ACTION_ADD_POLYGON,
+		ACTION_REMOVE_POLYGON,
+		ACTION_PAINT_WEIGHT,
+		ACTION_CLEAR_WEIGHT,
+		ACTION_MAX
+	};
 
 	Polygon2D *node = nullptr;
 
-	UVMode uv_mode;
-	AcceptDialog *uv_edit = nullptr;
-	Button *uv_button[UV_MODE_MAX];
+	VBoxContainer *polygon_edit = nullptr;
+	Mode current_mode;
+	Button *mode_buttons[MODE_MAX];
+	Action selected_action;
+	Button *action_buttons[ACTION_MAX];
 	Button *b_snap_enable = nullptr;
 	Button *b_snap_grid = nullptr;
-	Panel *uv_edit_background = nullptr;
+	MenuButton *edit_menu = nullptr;
+
+	Control *canvas = nullptr;
+	Panel *canvas_background = nullptr;
 	Polygon2D *preview_polygon = nullptr;
-	Control *uv_edit_draw = nullptr;
 	EditorZoomWidget *zoom_widget = nullptr;
-	HScrollBar *uv_hscroll = nullptr;
-	VScrollBar *uv_vscroll = nullptr;
-	MenuButton *uv_menu = nullptr;
+	HScrollBar *hscroll = nullptr;
+	VScrollBar *vscroll = nullptr;
 
-	Ref<ViewPanner> uv_panner;
-	void _uv_pan_callback(Vector2 p_scroll_vec, Ref<InputEvent> p_event);
-	void _uv_zoom_callback(float p_zoom_factor, Vector2 p_origin, Ref<InputEvent> p_event);
+	Ref<ViewPanner> panner;
+	void _pan_callback(Vector2 p_scroll_vec, Ref<InputEvent> p_event);
+	void _zoom_callback(float p_zoom_factor, Vector2 p_origin, Ref<InputEvent> p_event);
+	Vector2 draw_offset;
+	real_t draw_zoom;
```

Comment:
```suggestion
	real_t draw_zoom = 0;
```

Reply:
Is this something that should be applied to all properties of basic datatypes? For example, there's also `int point_drag_index; bool is_dragging; bool is_creating;` which aren't being assigned in the header.

Reply:
Yes.